### PR TITLE
automated: linux: add more options to docker networking

### PR DESCRIPTION
**Pull request by mwasilew**
Add "host" and "none" network options to docker-networking tests. This allows to cover all simple possibilities when it comes to docker networking. All tests can be executed in one run by adding -n option to the script invocation. Example:

    docker-networking.sh -n "bridge host"

Test names were appended with ${NETWORK} to indicate which network type is tested. Default is "bridge".

### CODE REVIEW

**Comment by mwasilew** [comment]
Example job https://lava.infra.foundries.io/scheduler/job/74256/definition#defline87

**Comment by mwasilew** [comment]
Fixed issues with skip_list and formatting. Also fixed the test for none network ping. Example run:
https://lava.infra.foundries.io/scheduler/job/74435

File: automated/linux/docker-networking/docker-networking.sh [hunk 1]
```diff
@@ -43,48 +50,83 @@ install_docker() {
...
+
+    skip_list=$(remove_one_from_skiplist "${skip_list}")
+    if [ -n "${HOST_IP}" ]; then
+        if [ "${NETWORK}" = none ]; then
+            # ping should fail with disabled networking
+            if ! docker run --name ping_localhost_host_network --network "${NETWORK}" --rm "${IMAGE}" ping -4 -c 4 "${HOST_IP}"; then
+                report_pass "docker-ping-host-network-${NETWORK}"
+            else
+                report_fail "docker-ping-host-network-${NETWORK}"
+            fi
```
Comment (roxell) [inline automated/linux/docker-networking/docker-networking.sh:120]: can this if else statement be written to check_return instead?

File: automated/linux/docker-networking/docker-networking.sh [hunk 2]
```diff
@@ -43,48 +50,83 @@ install_docker() {
...
+            check_return "docker-ping-host-network-${NETWORK}"
+        fi
+    else
+        report_skip "docker-ping-host-network-${NETWORK}"
+    fi
+done
 
-# IPv4 try pinging localhost from container with host networking
-docker run --name ping_localhost_host_network --rm -d "${IMAGE}" ping -4 -c 4 localhost
-check_return "docker-ping-localhost-host-network"
```
Comment (roxell) [inline automated/linux/docker-networking/docker-networking.sh:131]: maybe remove one extra newline?

File: automated/linux/docker-networking/docker-networking.sh (hunk 1 above)
Comment (mwasilew) [reply automated/linux/docker-networking/docker-networking.sh:120]: I'm expecting this to fail (ping shouldn't work with `none` networking). check_return will record fail where it should record pass.

File: automated/linux/docker-networking/docker-networking.sh (hunk 2 above)
Comment (mwasilew) [reply automated/linux/docker-networking/docker-networking.sh:131]: ok

File: automated/linux/docker-networking/docker-networking.sh (hunk 1 above)
Comment (roxell) [reply automated/linux/docker-networking/docker-networking.sh:120]: Could we do something like this in the check_return function?

`$ git df
diff --git a/automated/lib/sh-test-lib b/automated/lib/sh-test-lib
index bce0e716efa9..2dc14475cc5e 100755
--- a/automated/lib/sh-test-lib
+++ b/automated/lib/sh-test-lib
@@ -105,12 +105,13 @@ check_return() {
     [ "$#" -ne 1 ] && error_msg "Usage: check_return test_case"
     # shellcheck disable=SC2039
     local test_case="$1"
+    local xfail="${2:-}"
 
-    if [ "${exit_code}" -ne 0 ]; then
-        echo "${test_case} fail" | tee -a "${RESULT_FILE}"
+    if [ "${exit_code}" -ne 0 && -z "${xfail}" ]; then
+        report_fail "$1"
         return "${exit_code}"
     else
-        echo "${test_case} pass" | tee -a "${RESULT_FILE}"
+        report_pass "$1"
         return 0
     fi
 }
`

File: automated/linux/docker-networking/docker-networking.sh (hunk 2 above)
Comment (mwasilew) [reply automated/linux/docker-networking/docker-networking.sh]: done

File: automated/linux/docker-networking/docker-networking.sh (hunk 1 above)
Comment (mwasilew) [reply automated/linux/docker-networking/docker-networking.sh]: Added to sh-test-lib.

**Review by roxell — APPROVED** [review]
